third-party/jemalloc/README

```diff
@@ -0,0 +1,11 @@
+==========================
+jemalloc README for Chapel
+==========================
+
+This copy of jemalloc is being released with Chapel for convenience and
+uses a snapshot from:
```

Comment:
Oh!  Sorry, somehow I didn't catch that.  I think the term "snapshot" made me think we just picked up most current sources from master.  Although now that I think of it, that 4.0.4 version number does appear in this PR, in at least one place.  Must have been a brain cramp -- apologies.